crates/apollo-parser/benches/query.rs

```diff
@@ -0,0 +1,66 @@
+use apollo_parser::{ast, Lexer, LexerIterator};
+use criterion::*;
+
+fn parse_query(query: &str) {
+    let parser = apollo_parser::Parser::new(query);
+    let tree = parser.parse();
+
+    if !tree.errors().is_empty() {
+        panic!("error parsing query: {:?}", tree.errors());
+    }
+    let document = tree.document();
+
+    for definition in document.definitions() {
+        if let ast::Definition::OperationDefinition(operation) = definition {
+            let selection_set = operation
+                .selection_set()
+                .expect("the node SelectionSet is not optional in the spec; qed");
+            for selection in selection_set.selections() {
+                match selection {
+                    ast::Selection::Field(field) => {
+                        let _selection_set = field.selection_set();
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+}
+
+fn bench_query_parser(c: &mut Criterion) {
```

Comment:
oh! and thank you for all the extra benchmarks 🙏

Reply:
benchmarking is fun :)